oak_functions/loader/src/metrics.rs

```diff
@@ -244,13 +290,117 @@ impl PrivateMetricsProxy {
     /// aggregated metrics are returned.
     ///
     /// See [PrivateMetricsAggregator::report_metrics] for more details.
-    pub fn publish(self) -> Option<(usize, Vec<(String, i64)>)> {
+    fn publish(self) -> Option<(usize, Vec<(String, i64)>)> {
         if let Ok(mut aggregator) = self.aggregator.lock() {
             aggregator.report_metrics(self.data)
         } else {
             None
         }
     }
+
+    fn publish_metrics(self) {
```

Comment:
nit: perhaps move this function and the logger onto the `PrivateMetricsExtension` so that the proxy is only responsible for reporting metrics, not logging it?

Reply:
Good idea. Done.